kyaml/fn/framework/example_logger_injector_test.go

```diff
@@ -0,0 +1,63 @@
+package framework_test
+
+import (
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// In this example, we implement a function that injects a logger as a sidecar
+// container in workload APIs.
+
+func Example_loggeInjector() {
+	if err := command.AsMain(framework.ResourceListProcessorFunc(injectLogger)); err != nil {
+		os.Exit(1)
+	}
+}
+
+// injectLogger injects a logger container into the workload API resources.
+// generate implements the goframework.KRMFunction interface.
+func injectLogger(rl *framework.ResourceList) error {
+	var li LoggerInjection
+	if err := rl.FunctionConfig.As(&li); err != nil {
+		return err
+	}
+	for i, obj := range rl.Items {
+		if obj.GetApiVersion() == "apps/v1" && (obj.GetKind() == "Deployment" || obj.GetKind() == "StatefulSet" || obj.GetKind() == "DaemonSet" || obj.GetKind() == "ReplicaSet") {
```

Comment:
Same comment about taking advantage of the framework's matchers for this.

Btw, have you seen `TestTemplateProcessor_ContainerPatchTemplates_MultipleWorkloadKinds`, which shows how to use `TemplateProcessor` for a very similar use case? That processor handles the matter of where the container path is for you, to make it easy to support all the workload types.  It also makes it so you don't need to worry about the found/new case, since it uses SMP.